libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2316,17 +2335,32 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 	}
 	else
 	{
-		m_errorReporter.typeError(
-			5511_error,
-			arguments.front()->location(),
-			"Expected first argument to be a function pointer, not \"" +
-			type(*arguments.front())->humanReadableName() +
-			"\"."
-		);
-		return;
+		switch (_functionType->kind()) {
+			case FunctionType::Kind::ABIEncodeCall:
+				m_errorReporter.typeError(
+					5511_error,
+					arguments.front()->location(),
+					"Expected first argument to be a function pointer, not \"" +
+					type(*arguments.front())->humanReadableName() +
+					"\"."
+				);
+				return;
+			case FunctionType::Kind::ABIEncodeError:
+				m_errorReporter.typeError(
+					5512_error,
+					arguments.front()->location(),
+					"Expected first argument to be a custom error, not \"" +
```

Comment:
I was trying to think of some test cases and when I try something like:
```solidity
error E(uint a, uint b);
contract C {
    function g() public view returns (bytes memory) {
        bytes memory b1 = abi.encodeError(E(1, 2), (1, 2));
        return b1;
    }
}
```
The following error message is generated:
`Expected first argument to be a custom error, not "error".`
Which is a bit weird. Maybe we should be more precise and change to `[...] custom error name [...]` ?